mm2src/coins/eth.rs

```diff
@@ -762,6 +765,23 @@ struct SavedErc20Events {
     latest_block: U64,
 }
 
+/// Specifies which blockchain the EthCoin operates on: EVM-compatible or TRON.
+/// This distinction allows unified logic for EVM & TRON coins.
+#[derive(Clone, Debug)]
+pub enum ChainSpec {
+    Evm { chain_id: u64 },
+    Tron { network: tron::Network },
+}
+
+impl ChainSpec {
+    pub fn chain_id(&self) -> Option<u64> {
+        match self {
+            ChainSpec::Evm { chain_id } => Some(*chain_id),
+            ChainSpec::Tron { .. } => None,
+        }
+    }
+}
+
```

Comment:
a quick question: why no `chain_id` if `TRON` itself uses/has `chain_id` https://developers.tron.network/reference/eth_chainid?

Reply:
That's not really the case, this is only in Tron JSON RPC which doesn't have the full methods needed like a method for broadcasting the transaction ref. https://github.com/KomodoPlatform/komodo-defi-framework/issues/1542#issuecomment-1493385418
I guess Tron's JSON RPC is primarily used for compatibility with Ethereum-based tools so we will probably need it when implementing WalletConnect for Tron, but let's not have it for now and add it when it's needed.

Reply:
Thanks!

Reply:
> I guess Tron's JSON RPC is primarily used for compatibility with Ethereum-based tools so we will probably need it when implementing WalletConnect for Tron

This info gives me greater confidence that implementing TRON as part of `EthCoin` is the right approach instead of treating it as an entirely separate protocol.